### PR TITLE
InteractiveViewer mouse scale bug

### LINKED ISSUE #59411: InteractiveViewer ignores scaleEnabled on web
## Steps to Reproduce

1. Use `InteractiveViewer` in a Flutter web app.
2. Specify `scaleEnabled: false`.
3. Run the Flutter web app.
4. Use the mouse scroll wheel when the cursor is on the widget.
→ viewer zooms in, even though it should not

```dart
InteractiveViewer(
  scaleEnabled: false,
  minScale: 1,
  maxScale: 1,
  child: Text('text'),
)
```

@justinmc

### PR DESCRIPTION
**Pull request by justinmc**
## Description

InteractiveViewer wasn't respecting the `scaleEnabled` parameter when scaling was done with a mouse.

## Related Issues

Closes https://github.com/flutter/flutter/issues/59411

## Tests

I added the following tests:

  * Test that scaling with the mouse works normally, and test that it doesn't work when `scaleEnabled` is false.

## Breaking Change

None

### CODE REVIEW

File: packages/flutter/lib/src/widgets/interactive_viewer.dart [hunk 1]
```diff
@@ -659,13 +659,13 @@ class _InteractiveViewerState extends State<InteractiveViewer> with TickerProvid
 
   // Returns true iff the given _GestureType is enabled.
   bool _gestureIsSupported(_GestureType gestureType) {
-    if (_gestureType == _GestureType.pan && !widget.panEnabled) {
+    if (gestureType == _GestureType.pan && !widget.panEnabled) {
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/widgets/interactive_viewer.dart:662]: nit: maybe use `switch` instead?

**Review by LongCatIsLooong — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/widgets/interactive_viewer.dart (hunk 1 above)
Comment (justinmc) [reply packages/flutter/lib/src/widgets/interactive_viewer.dart]: That seems simpler to me, will do 👍

Comment (LongCatIsLooong) [reply packages/flutter/lib/src/widgets/interactive_viewer.dart]: I think using `default` clause is discouraged.

**Pull request merged**